---
title: "BWG"
author: "Nathan Wyatt"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Introduction
Welcome to week 1 of the Bioinformatics Working Group. My goal for this first
meetup is to cover:
  - some SciNet Basics
  - An introdcution to R/Rstudio 
  - Demonstrate the utility of ChatGPT for learning 

#SciNet: What is it and how to get started
SciNet is the term applied to the high performance computing cluster that the 
USDA has spent a significant amount of money developing. 

It consists of two separate but connected clusters:
  - Ceres - located in Ames Iowa
  - Atlas - located in Starkville Mississippi
  
I primarily use Ceres. There are some subtle differences between the inner 
workings of Ceres and Atlas but in large part they are similar and differences
are well documented.

All SciNet documents can be found at: https://scinet.usda.gov/

##Who can use SciNet?
The short answer is everyone. SciNet does not requrie USDA credentials and so
it is already available for everyone in this room. However, if you do not have 
a USDA email address the access must be requested by your collaborator or SY.

The link to request an account can be found here: https://scinet.usda.gov/about/signup

##How should I log in to SciNet?
The answer is a program called Putty... and you will need local IT to install it. 

##Why should you use SciNet?
In general, SciNet provides a computational space for us all to work without the 
need for IT staff to install and direct our software usage. This means that anything
you want to accomplish is possible as long as you have the know-how. SciNet also
comes with a ton of pre-installed programs that are common to bioinformatics.
  Examples:
    ncbi-blast+ (for blasting)
    samtools (for sequence alignment files)
    bcftools (variant calling and manipulation)
    bedtools (the swiss army knife of genomics)
    seqtk (sequence manipulations)
    GATK (variant calling)
    mummer (whole genome alignment)
    spades (short read genome assembler)





## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
